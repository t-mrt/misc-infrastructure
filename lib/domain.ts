import * as cdk from "@aws-cdk/core"
import * as r from "@aws-cdk/aws-route53"
import * as t from "@aws-cdk/aws-route53-targets"
import * as c from "@aws-cdk/aws-cloudfront"

// TODO: divide stack by hosted zone

export class DomainStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props)
        const zone = new r.PublicHostedZone(this, "yux3", {
            zoneName: "yux3.net",
        })
        new r.CnameRecord(this, "blog", {
            zone: zone,
            domainName: "hatenablog.com.",
            recordName: "blog",
        })
        const distribution = c.Distribution.fromDistributionAttributes(this, "ImportedDist", {
            domainName: "d25ymprel65g4i.cloudfront.net",
            distributionId: "E1AB8RFEZ1H7F0",
        })
        new r.ARecord(this, "img", {
            zone: zone,
            target: r.RecordTarget.fromAlias(new t.CloudFrontTarget(distribution)),
            recordName: "img",
        })
    }
}
