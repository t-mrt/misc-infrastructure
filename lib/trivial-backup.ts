import * as cdk from "@aws-cdk/core"
import * as iam from "@aws-cdk/aws-iam"
import * as s3 from "@aws-cdk/aws-s3"

export class TrivialBackupStack extends cdk.Stack {
    constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
        super(scope, id, props)
        const user = new iam.User(this, "User", {
            userName: "trivial_backup",
        })

        const bucket = new s3.Bucket(this, "Bucket", {
            versioned: true,
            bucketName: "trivial-backup-02",
            blockPublicAccess: s3.BlockPublicAccess.BLOCK_ALL,
        })

        bucket.addToResourcePolicy(
            new iam.PolicyStatement({
                actions: ["s3:PutObject"],
                resources: [bucket.arnForObjects("*")],
                principals: [user],
            })
        )
    }
}
