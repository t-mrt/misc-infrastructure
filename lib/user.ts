import * as cdk from '@aws-cdk/core';
import * as iam from '@aws-cdk/aws-iam'

export class UserStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);
    const user = new iam.User(this, 'GeneralPurposeUser', {
      userName: "t_"
    });

    const role = new iam.Role(this, 'PowerUserRole', {
      assumedBy: new iam.ArnPrincipal(user.userArn).withConditions({
        Bool: {
          "aws:MultiFactorAuthPresent": "true"
        },
      }),
      roleName: "admin_role",
    });

    user.addToPolicy(new iam.PolicyStatement({
      resources: [role.roleArn],
      actions: ['sts:AssumeRole'],
    }))

    role.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName("AdministratorAccess"))
  }
}
