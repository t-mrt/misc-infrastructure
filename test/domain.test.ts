import * as cdk from "@aws-cdk/core"
import { SynthUtils } from "@aws-cdk/assert"
import * as d from "../lib/domain"

test("snapshot", () => {
    const app = new cdk.App()
    const stack = new d.DomainStack(app, "MyTestStack")
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot()
})
