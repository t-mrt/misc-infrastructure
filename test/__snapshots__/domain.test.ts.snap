// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Object {
  "Mappings": Object {
    "AWSCloudFrontPartitionHostedZoneIdMap": Object {
      "aws": Object {
        "zoneId": "Z2FDTNDATAQYW2",
      },
      "aws-cn": Object {
        "zoneId": "Z3RFFRIM2A3IF5",
      },
    },
  },
  "Resources": Object {
    "blogF3DAE94C": Object {
      "Properties": Object {
        "HostedZoneId": Object {
          "Ref": "yux326B34FD5",
        },
        "Name": "blog.yux3.net.",
        "ResourceRecords": Array [
          "hatenablog.com.",
        ],
        "TTL": "1800",
        "Type": "CNAME",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "img31CD142E": Object {
      "Properties": Object {
        "AliasTarget": Object {
          "DNSName": "d25ymprel65g4i.cloudfront.net",
          "HostedZoneId": Object {
            "Fn::FindInMap": Array [
              "AWSCloudFrontPartitionHostedZoneIdMap",
              Object {
                "Ref": "AWS::Partition",
              },
              "zoneId",
            ],
          },
        },
        "HostedZoneId": Object {
          "Ref": "yux326B34FD5",
        },
        "Name": "img.yux3.net.",
        "Type": "A",
      },
      "Type": "AWS::Route53::RecordSet",
    },
    "yux326B34FD5": Object {
      "Properties": Object {
        "Name": "yux3.net.",
      },
      "Type": "AWS::Route53::HostedZone",
    },
  },
}
`;
